### DISCUSSION TITLE
Charger les emojis

### DISCUSSION THREAD

**Discussion by camilledesvigne** [boisgera/python-advanced-eval#2] [Q&A]
Bonsoir Monsieur,

je n’arrive pas à charger les emojis correctement pour le projet, y compris après des recherches internet. Auriez-vous une indication quant à la méthode à utiliser s’il vous plaît ?
Bonne soirée,

Camille Desvigne

**Comment by boisgera** [comment]
Bonsoir Camille,

Est-ce que tu pourrais me donner qq détails ? Les émojis dont tu parles sont ceux des notebooks de référence du projet ? Et ils sont incorrectement représentés quand tu ouvres un notebook avec jupyter notebook ? Ou est-ce qu'il s'agit d'autre chose ?

**Reply by boisgera** [reply]
(En supposant que j'aie bien interprété le problème, cf https://github.com/boisgera/python-advanced-eval/discussions/2#discussioncomment-1844281.)

Par acquis de conscience, j'ai testé l'affichage des notebooks de référence sur un Windows 10 fraîchement installé (VirtualBox).
J'ai fait une installation "standard" (Anaconda, VS Code + extension Python, Git Bash). J'ai ensuite vérifié que les émojis des notebooks étaient bien lisibles dans 3 configurations :

  - dans l'éditeur de notebook VS Code,

  - avec `jupyter notebook` lancé dans le terminal git-bash,

  - avec `jupyter notebook` lancé dans le "Command Prompt" (`cmd.exe`) Windows.

⚠️ Je remarque que dans les trois contextes que j'ai testé (4 avec mon Linux), l'encodage par défaut était bien `utf-8`, ce qui est nécessaire pour que les émojis soient correctement interprétés (les fichiers notebooks fournis sont encodés en utf-8). Je suppose qu'il en sera de même sur Mac, mais pas forcément sur des Windows plus anciens.

Pour vérifier que cela est bien le cas sur vos machines, affichez l'encodage utilisé par défaut :  

  - Si vous êtes dans un notebook (ou un programme Python) avec des émojis mal représentés, exécutez :
   
    ``` python
    import sys
    print(sys.getdefaultencoding())
    ```

  - ou, si vous êtes dans le terminal ou vous lancez le notebook jupyter, exécutez :

    ```
    python -c "import sys; print(sys.getdefaultencoding())"
    ```

Si ce n'est pas utf-8 qui s'affiche, le problème est sans doute là. Il faudra alors rechercher un moyen de basculer votre système vers utf-8.
